### PR TITLE
limit range e2e flake: Adding some debug messages on test failure

### PR DESCRIPTION
**Pull request by nikhiljindal**
To help in debugging https://github.com/kubernetes/kubernetes/issues/21234

### CODE REVIEW

File: test/e2e/limit_range.go [hunk 1]
```diff
@@ -61,7 +61,11 @@ var _ = Describe("LimitRange", func() {
 		Expect(err).NotTo(HaveOccurred())
 		for i := range pod.Spec.Containers {
 			err = equalResourceRequirement(expected, pod.Spec.Containers[i].Resources)
-			Expect(err).NotTo(HaveOccurred())
+			if err != nil {
+				// Print the pod to help in debugging.
+				Logf("Pod %+v does not have the expected requirements", pod)
```
Comment (ihmccreery) [inline test/e2e/limit_range.go]: Would it make more sense for the error returned by `equalResourceRequirement` to have the necessary information?

Comment (nikhiljindal) [inline test/e2e/limit_range.go]: equalResourceRequirement() does not get the pod. It only gets the expected and actual requirements which it anyways prints in Logf().
I want to see if other fields in the pod are being set or not.

**Comment by nikhiljindal** [comment]
@k8s-bot  test this please github issue: https://github.com/kubernetes/kubernetes/issues/19732

**Comment by nikhiljindal** [comment]
@k8s-bot test this issue: https://github.com/kubernetes/kubernetes/issues/21234

File: test/e2e/limit_range.go (hunk 1 above)
Comment (ihmccreery) [inline test/e2e/limit_range.go]: Ack.

**Comment by nikhiljindal** [comment]
k8s-bot test this issue: https://github.com/kubernetes/kubernetes/issues/20096

**Comment by nikhiljindal** [comment]
@k8s-bot test this issue: #20096

**Comment by wojtek-t** [comment]
adding P0 to have it merged sooner to get debuggin info

**Pull request merged**